### PR TITLE
Added support for names in Text Splitting functions.

### PR DESCRIPTION
**Pull request by Cyberghxst**
## Type
- [x] Functions: **Text Split functions**

Dependencies (Third Party Modules) needed: None

Want a credit? Discord tag or other social media link: @cyberghxst

Referenced Issue: #NaN

## Description
According to a suggestion on the Discord server, I added name support for text splitting functions that are **NOT BREAKING** due to the positioning of the new optional name.
Before submitting this pull request, I did some tests using the following code:
```js
bot.awaitedCommand({
    name: 'venecia',
    code: `
        $log[hello fro venecia command: $message[1]]
    `
})
bot.awaitedCommand({
    name: 'colombo',
    code: `
        $log[hello FROMM colombo command: $message[1]]
    `
})
bot.awaitedCommand({
    name: 'list',
    code: `
        $log[hello FROMM LIST command: $message[1]]
    `
})

bot.readyCommand({
    code: `
        $textSplitMap[venecia;colombo;list;meals]
        $log[$joinSplitText[ - ;meals]]
        $removeTextSplitElement[Carne polaca;meals]
        $log[$joinSplitText[ - ;meals]]
        $removeSplitTextElement[Pizza;meals]
        $log[$joinSplitText[ - ;meals]]
        $editTextSplitElement[3;Patatas;meals]
        $log[$findTextSplitIndex[Papas;meals]]
        $log[$joinSplitText[ - ;meals]]
        $concatTextSplit[Hotdogs;Carne polaca;Pizza;meals]
        $log[$splitText[3;meals]]
        $log[$filterTextSplitElement[P;starts; -uwu- ;meals]]
        $log[$getTextSplitLength[meals]]
        $textSplit[Carne,Ensalada,Papas,Hamburguesa;,;meals]

        $log[\n\n\n]

        $textSplitMap[venecia;colombo;list]
        $log[$joinSplitText[ - ]]
        $removeTextSplitElement[Chango]
        $log[$joinSplitText[ - ]]
        $removeSplitTextElement[Zanate]
        $log[$joinSplitText[ - ]]
        $editTextSplitElement[3;Michu]
        $log[$findTextSplitIndex[Gato]]
        $log[$joinSplitText[ - ]]
        $concatTextSplit[Serpiente;Mono;Chango;Zanate]
        $log[$splitText[3]]
        $log[$filterTextSplitElement[P;starts; -uwu- ]]
        $log[$getTextSplitLength]
        $textSplit[Perro,Pato,Gato,Pájaro,Gusano;,]
    `
})
```
Getting the expected output:
SCR1
SCR2

I hope this gets accepted. If you have any comments or suggestions, feel free to tell me.

### CODE REVIEW

File: src/core/interpreter.js [hunk 1]
```diff
@@ -87,7 +87,7 @@ const Interpreter = async (
                 data.vars || {},
                 data.object || {},
                 ["roles", "users", "everyone"],
-                data.array || [],
```
Comment (Leref) [inline src/core/interpreter.js]: You're changing into object, this doesn't make sense.

File: src/functions/misc/editTextSplitElement.js [hunk 2]
```diff
@@ -2,7 +2,7 @@ module.exports = async (d) => {
     const data = d.util.aoiFunc(d);
     if (data.err) return d.error(data.err);
 
-    let [index, text] = data.inside.splits;
+    let [index, text, name = 'main'] = data.inside.splits;
```
Comment (Leref) [inline src/functions/misc/editTextSplitElement.js]: There shouldn't be default of main at all.

File: src/functions/misc/removeSplitTextElement.js [hunk 3]
```diff
@@ -2,13 +2,16 @@ module.exports = d => {
     const data = d.util.aoiFunc(d);
     if (data.err) return d.error(data.err);
 
-    const [...elements] = data.inside.splits;
+    let elements = data.inside.splits, name = elements.length < 1 ? 'main' : elements.pop()
+
+    // If the split array is not found, returns the command name into the list.
+    if (!(name in d.array)) elements.push(name), name = 'main';
```
Comment (Leref) [inline src/functions/misc/removeSplitTextElement.js]: It was fine how it was.

**Review by Leref — CHANGES_REQUESTED** [review]
Required.

File: src/core/interpreter.js (hunk 1 above)
Comment (Cyberghxst) [reply src/core/interpreter.js]: It is since the data inside d.array will be arrays.

**Pull request closed without merging**